clang/runtime/dpct-rt/include/dpl_extras/algorithm.h.inc

```diff
@@ -1531,29 +1532,26 @@ inline void segmented_sort_pairs_by_two_pair_sorts(
     // coordinate to mark segments
     policy.queue()
         .submit([&](sycl::handler &h) {
-          h.parallel_for(sycl::nd_range<1>{work_group_size, work_group_size},
-                         ([=](sycl::nd_item<1> item) {
-                           auto sub_group = item.get_sub_group();
-                           ::std::size_t num_subgroups =
-                               sub_group.get_group_range().size();
-                           ::std::size_t local_size =
-                               sub_group.get_local_range().size();
-
-                           ::std::size_t sub_group_id =
-                               sub_group.get_group_id();
-                           while (sub_group_id < nsegments) {
-                             ::std::size_t subgroup_local_id =
-                                 sub_group.get_local_id();
-                             std::size_t i = begin_offsets[sub_group_id];
-                             std::size_t end = end_offsets[sub_group_id];
-                             while (i + subgroup_local_id < end) {
-                               segments[i + subgroup_local_id] =
-                                   sub_group_id;
-                               i += local_size;
-                             }
-                             sub_group_id += num_subgroups;
-                           }
-                         }));
+          h.parallel_for(
+              sycl::nd_range<1>{work_group_size, work_group_size},
+              ([=](sycl::nd_item<1> item) {
+                auto sub_group = item.get_sub_group();
+                ::std::size_t num_subgroups =
+                    sub_group.get_group_range().size();
+                ::std::size_t local_size = sub_group.get_local_range().size();
+
+                ::std::size_t sub_group_id = sub_group.get_group_id();
+                while (sub_group_id < nsegments) {
+                  ::std::size_t subgroup_local_id = sub_group.get_local_id();
```

Comment:
I think here don't need `::` before std?

Reply:
We use it to ensure that we are picking up the `std` from the global namespace, `::std` is more explicit in meaning than `std`.  You are correct that it very likely wont many any difference here in practice here, but it should not hurt us.